src/AutoRest.CSharp/MgmtTest/Extensions/CodeWriterExtensions.cs

```diff
@@ -104,12 +105,50 @@ private static CodeWriter AppendFrameworkTypeValue(this CodeWriter writer, CShar
             if (type.FrameworkType == typeof(BinaryData))
                 return writer.AppendBinaryData(exampleValue);
 
+            if (type.FrameworkType == typeof(DataFactoryElement<>))
+                return writer.AppendDataFactoryElementValue(type, exampleValue);
+
             if (exampleValue.Schema is ObjectSchema objectSchema)
                 return writer.AppendComplexFrameworkTypeValue(objectSchema, type.FrameworkType, exampleValue);
 
             return writer.AppendRawValue(exampleValue.RawValue, type.FrameworkType, exampleValue.Schema.Type);
         }
 
+        private static CodeWriter AppendDataFactoryElementValue(this CodeWriter writer, CSharpType type, ExampleValue exampleValue)
+        {
+            if (type.FrameworkType != typeof(DataFactoryElement<>))
+                throw new ArgumentException("DataFactoryElement<> is expected but got: " + type.ToString());
+
+            const string DFE_OBJECT_SCHEMA_PREFIX = "DataFactoryElement-";
+
+            if (exampleValue.Schema is ObjectSchema os && os.Name.StartsWith(DFE_OBJECT_SCHEMA_PREFIX))
+            {
+                const string DFE_OBJECT_PROPERTY_TYPE = "type";
+                const string DFE_OBJECT_PROPERTY_VALUE = "value";
+                Dictionary<string, string> DFE_FACTORY_METHOD_MAPPING = new Dictionary<string, string>()
+                {
+                    ["Expression"] = "FromExpression",
+                    ["SecureString"] = "FromSecretString",
+                    ["AzureKeyVaultSecretReference"] = "FromKeyVaultSecretReference"
+                };
+
+                string dfeType = (string)exampleValue.Properties![DFE_OBJECT_PROPERTY_TYPE].RawValue!;
+                string dfeValue = (string)exampleValue.Properties![DFE_OBJECT_PROPERTY_VALUE].RawValue!;
+
+                writer.UseNamespace(type.Namespace!);
+                writer.AppendRaw($"{type.ToString().Trim()}.{DFE_FACTORY_METHOD_MAPPING[dfeType]}(");
```

Comment:
Please use
```suggestion
                writer.AppendRaw($"{type}.{DFE_FACTORY_METHOD_MAPPING[dfeType]}(");
```
instead.
Also it would be more readable if we could first get the method name
```csharp
var dfeMethodName = DFE_FACTORY_METHOD_MAPPING[dfeType];
```
and then use it in the interpolation